### ISSUE TITLE
BatchNorm usage

### ISSUE THREAD

**Issue by Eniac-Xie** [isht7/pytorch-deeplab-resnet#15]
Hi, the parameters of BatchNorm layer in resnet101 is fixed by [here](https://github.com/isht7/pytorch-deeplab-resnet/blob/master/deeplab_resnet.py#L60)

 But the running_mean and running_var is also need to be fix, so I think we need to set BatchNorm to eval mode, not just fix parameters (weight and bias)

**Comment by isht7** [comment]
The model is trained in eval mode as can be seen [here](https://github.com/isht7/pytorch-deeplab-resnet/blob/master/train.py#L200).

**Comment by Eniac-Xie** [comment]
Thank you for your reminder , so the code [here](https://github.com/isht7/pytorch-deeplab-resnet/blob/master/deeplab_resnet.py#L61) is redundant?

**Comment by isht7** [comment]
No, it is not redundant. model.eval() is for keeping running mean and running variance fixed. Setting requires_grad = False is for fixing the gamma and beta parameters in the batch-norm. Read more about batchnorm [here](https://kratzert.github.io/2016/02/12/understanding-the-gradient-flow-through-the-batch-normalization-layer.html)

**Comment by pkuCactus** [comment]
Hi, i don't understand why fix the batchnormal parameters?

**Comment by isht7** [comment]
This is a re-implementation of deeplab-resent. The original code fixed batchnorm parameters and therefore we also did that. They might have done this to reduce the number of trainable parameters because the number of training images is quite less. This could help prevent overfitting.
